File: c++/Aksesoris.cpp
#include "Petshop.cpp"

class Aksesoris : public Petshop{
    protected:
    string jenis;
    string bahan;
    string warna;

    
    //constructor
    Aksesoris(){
        this->jenis = "";
        this->bahan = "";
        this->warna = "";
    }
 
    Aksesoris(string jenis, string bahan, string warna, string namaProduk, int hargaProduk, int stokProduk) : Petshop(namaProduk, hargaProduk, stokProduk){
        this->jenis = jenis;
        this->bahan = bahan;
        this->warna = warna;
    }
    
    public:
    //getter
    string get_jenis() const { return this->jenis; }
    string get_bahan() const { return this->bahan; }
    string get_warna() const { return this->warna; }

    //setter
    void set_jenis(string jenis) { this->jenis = jenis; }
    void set_bahan(string bahan) { this->bahan = bahan; }
    void set_warna(string warna) { this->warna = warna; }

    //destructor
    ~Aksesoris(){
        
    };
};

File: php/Aksesoris.php
<?php
require_once "Petshop.php";

class Aksesoris extends Petshop{
    protected $jenis;
    protected $bahan;
    protected $warna;

    public function __construct($jenis, $bahan, $warna, $produk, $harga, $stok, $foto){
        parent::__construct($produk, $harga, $stok, $foto);
        $this->jenis = $jenis;
        $this->bahan = $bahan;
        $this->warna = $warna;
    }


    public function getJenis() {
        return $this->jenis;
    }
    public function getBahan() {
        return $this->bahan;
    }
    public function getWarna() {
        return $this->warna;
    }
    
    public function setJenis($jenis) {
        $this->jenis = $jenis;
    }
    public function setBahan($bahan) {
        $this->bahan = $bahan;
    }
    public function setWarna($warna) {
        $this->warna = $warna;
    }
}

?>